sql/core/src/main/scala/org/apache/spark/sql/api/r/SQLUtils.scala

```diff
@@ -158,7 +158,7 @@ private[sql] object SQLUtils extends Logging {
     val dis = new DataInputStream(bis)
     val num = SerDe.readInt(dis)
     Row.fromSeq((0 until num).map { i =>
-      doConversion(SerDe.readObject(dis), schema.fields(i).dataType)
+      doConversion(SerDe.readObject(dis, jvmObjectTracker = null), schema.fields(i).dataType)
```

Comment:
I think this one is safe because this is called on the output from the R UDF and the R UDF is assumed to return native types (i.e. no references to other JVM objects)